smarts/core/smarts.py

```diff
@@ -312,7 +326,26 @@ def done_agent_ids(dones):
         self._teardown_vehicles(vehicles_to_teardown)
 
     def reset(self, scenario: Scenario):
-        if scenario == self._scenario and self._reset_agents_only:
+        tries = 2
+        first_exception = None
```

Comment:
Whats the reason behind 2 tries?

Reply:
Situational problems occur within the reset sometimes, I considered more than 1 retry but then there is no particular number that will work if the engine is failing. My reasoning here is that reset should not fail for an edge case on `reset` but I would say 
`reset` failure twice in a row indicates it is not an edge case.